examples/tutorials/introduction/3_observation.py

```diff
@@ -0,0 +1,193 @@
+"""
+Observation
+===========
+
+In our code so far, there is a problem that it is possible for certain related
+values to get out of sync with one another.  For example, if we change the
+filename after we have read the image into memory, then the data in memory
+still refers to the old image.  It would be nice if we could automatically
+re-load the image if the filename changes.  Traits allows you to do this.
+
+The Observe Decorator
+---------------------
+
+We want to have the `read_image` method run whenever the ``filename`` trait
+changes.  We can do this by adding an ``observe`` decorator to the method::
+
+
+    class Image(HasTraits):
+        ...
+
+        @observe('filename')
+        def read_image(self):
+            ...
+
+For most traits, this will run only when the trait's value *actually* changes,
+not just when the value is set.  So if you do::
+
+    >>> image.filename = "sample_0001.png"
+    >>> image.filename = "sample_0001.png"
+
+then the code will only be run once.
+
+You can still call the ``read_image`` function if you want to, but now you
+shouldn't need to.
+
+Observing Multiple Traits
+-------------------------
+
+If you look at the computation of ``pixel_area`` in the original code, it
+looks like this::
+
+    self.pixel_area = self.scan_height * self.scan_width / self.image.size
+
+It depends on the ``scan_width``, ``scan_height`` and the ``image``, so we
+would like to listen to changes to *all* of these.  We could write three
+``@observe`` functions, one for each trait, but the content would be the
+same for each.  A better way to do this is to have the observer listen to
```

Comment:
Aha, the new `@observe` can be called many times:
```
@observe("scan_width")
@observe("scan_height")
@observe("image")
def update_pixel_area(self, event):
    ....
```

But that can't be shown with the aliased `on_trait_change`.  The old `on_trait_change` supports a list of str as the input as well: `["scan_width", "scan_height", "image"]`, and so does the new `observe`  in order to make migration easier. Which style to go for is a bit subjective anyway... just wanted to mention :)